safe/common/parameters/group_select_parameter_widget.py

```diff
@@ -0,0 +1,223 @@
+
+        :param parameter: A GroupSelectParameter object.
+        :type parameter: GroupSelectParameter
+        """
+        QWidget.__init__(self, parent)
+        self._parameter = parameter
+
+        # Store spin box
+        self.spin_boxes = {}
+
+        # Create elements
+        # Label (name)
+        self.label = QLabel(self._parameter.name)
+
+        # Layouts
+        self.main_layout = QVBoxLayout()
+        self.input_layout = QVBoxLayout()
+
+        # _inner_input_layout must be filled with widget in the child class
+        self.inner_input_layout = QVBoxLayout()
+
+        self.radio_button_layout = QGridLayout()
+
+        # Create radio button group
+        self.input_button_group = QButtonGroup()
+
+        # List widget
+        self.list_widget = QListWidget()
+        self.list_widget.setSelectionMode(QAbstractItemView.ExtendedSelection)
+        self.list_widget.setDragDropMode(QAbstractItemView.DragDrop)
+        self.list_widget.setDefaultDropAction(Qt.MoveAction)
+        self.list_widget.setEnabled(False)
+        self.list_widget.setSizePolicy(
+            QSizePolicy.Maximum, QSizePolicy.Expanding)
+
+        for i, key in enumerate(self._parameter.options):
+            value = self._parameter.options[key]
+            radio_button = QRadioButton(value.get('label'))
+            self.radio_button_layout.addWidget(radio_button, i, 0)
+            if value.get('type') == SINGLE_DYNAMIC:
+                double_spin_box = QDoubleSpinBox()
+                self.radio_button_layout.addWidget(double_spin_box, i, 1)
+                double_spin_box.setValue(value.get('value', 0))
+                double_spin_box.setMinimum(
+                    value.get('constraint', {}).get('min', 0))
+                double_spin_box.setMaximum(value.get(
+                    'constraint', {}).get('max', 1))
+                double_spin_box.setSingleStep(
+                    value.get('constraint', {}).get('step', 0.01))
+                step = double_spin_box.singleStep()
+                if step > 1:
+                    precision = 0
+                else:
+                    precision = len(str(step).split('.')[1])
+                    if precision > 3:
+                        precision = 3
+                double_spin_box.setDecimals(precision)
+                self.spin_boxes[key] = double_spin_box
+
+                # Enable spin box depends on the selected option
+                if self._parameter.selected == key:
+                    double_spin_box.setEnabled(True)
+                else:
+                    double_spin_box.setEnabled(False)
+
+            elif value.get('type') == STATIC:
+                static_value = value.get('value', 0)
+                if static_value is not None:
+                    self.radio_button_layout.addWidget(
+                        QLabel(str(static_value)), i, 1)
+            elif value.get('type') == MULTIPLE_DYNAMIC:
+                selected_fields = value.get('value', [])
```

Comment:
__[flake8]__
* [F841] local variable 'selected_fields' is assigned to but never used

<!-- Comment by [SideCI](https://sideci.com) -->